classes/class-p4-master-site.php

```diff
@@ -674,9 +674,14 @@ public function p4_auto_generate_excerpt( $post_id, $post ) {
 			remove_action( 'save_post', [ $this, 'p4_auto_generate_excerpt' ], 10 );
 
 			// Generate excerpt text.
-			$post_excerpt   = strip_shortcodes( $post->post_content );
-			$post_excerpt   = apply_filters( 'the_content', $post_excerpt );
-			$post_excerpt   = str_replace( ']]>', ']]&gt;', $post_excerpt );
+			preg_match( "'<p>(.*?)</p>'", $post->post_content, $match_paragraph );
```

Comment:
Seems to make sense to do this. Just checking, are we sure there are no cases where there could be text in something else than a `<p>` tag that we still want to be part of the excerpt?

Reply:
Shouldn't this regex just be inside single quotes? Currently it will try to match everything that is inside the first `<p>` tag that is inside single quotes.

Reply:
There could be. We can have a look at the other blocks and what they produce.